esp-bootloader-esp-idf/src/partitions.rs

```diff
@@ -0,0 +1,757 @@
+//! # Partition Table Support
+//!
+//! ## Overview
+//!
+//! This module allows reading the partition table and conveniently
+//! writing/reading partition contents.
+//!
+//! For more information see <https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/partition-tables.html#built-in-partition-tables>
+
+use embedded_storage::Region;
+
+/// Maximum length of a partition table.
+pub const PARTITION_TABLE_MAX_LEN: usize = 0xC00;
+
+const PARTITION_TABLE_OFFSET: u32 =
+    esp_config::esp_config_int!(u32, "ESP_BOOTLOADER_ESP_IDF_CONFIG_PARTITION_TABLE_OFFSET");
+
+const RAW_ENTRY_LEN: usize = 32;
+const ENTRY_MAGIC: u16 = 0x50aa;
+const MD5_MAGIC: u16 = 0xebeb;
+
+/// Represents a single partition entry.
+#[repr(C)]
+pub struct PartitionEntry<'a> {
+    binary: &'a [u8],
```

Comment:
Might be worth putting a size in here, so the compiler can (more easily) elide some bound checks.
```suggestion
    binary: &'a [u8; RAW_ENTRY_LEN],
```